src/adc.rs

```diff
@@ -375,9 +403,7 @@ macro_rules! adc_hal {
                     type Error = ();
 
                     fn read(&mut self, _pin: &mut PIN) -> nb::Result<WORD, Self::Error> {
-                        self.power_up();
```

Comment:
If I understand correctly,  we opt to sacrifice power consumption for accuracy here. Right ? Alternatively we could calibrate before each read, but that might be tooooo slow...

Reply:
That is correct, also, that was the reason I created the split method, so if you want to save power you could use it only in the time you needed it, in contrast to have to wait 10us+ every conversion.

And as a side note, I think that the first time you would call the old method it would cause some problem, because the adc was already up, so if you set the ADON bit you would start a conversion.

Reply:
I confirm the observation regarding the first measurement in the previous approach. However with new approach (w/o power_up/power_down around convert) _all_ the values look strange, not only the first one. At least this is the case for temperature sensor. I need to do more testing using voltage inputs...

Reply:
I did more tests with various ADC clock settings. It turns out that sampling settings for temperature sensor (suggested in my previous PR #67 ) should be modified as follows to work well with introduced delays:

```
diff --git a/src/adc.rs b/src/adc.rs
index b1a73c2..d4752a9 100644
--- a/src/adc.rs
+++ b/src/adc.rs
@@ -468,7 +468,7 @@ impl Adc<ADC1> {
             2_400_001 ... 3_100_000 => AdcSampleTime::T_28,
             3_100_001 ... 4_000_000 => AdcSampleTime::T_41,
             4_000_001 ... 5_000_000 => AdcSampleTime::T_55,
-            5_000_001 ... 10_000_000 => AdcSampleTime::T_71,
+            5_000_001 ... 14_000_000 => AdcSampleTime::T_71,
             _ => AdcSampleTime::T_239,
         };
```

Could you please meld this fixup into you PR ?

Reply:
Done, thank you for helping out :)